pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -131,11 +136,11 @@ checkout_hash(){
     local ref="$2"
 
     if test -z "$hash"; then
-        hash=$(hash_from_ref $ref)
+        hash=$(hash_from_ref "$ref")
     fi
 
     git fetch -t ${builder:+--progress} origin || return 1
-    git checkout -b $branchName $hash || return 1
+    git checkout -b "$branchName" "$hash" || return 1
```

Comment:
Adding double quote does not serve any purpose as the shell expand its arguments as one unsplitable value, which is a different story when the argument is re-interpreted.  You can test it in you shell:

```
$ foo="test arg"
$ set -- $foo numbering
$ echo $1
test arg
$ echo $2
numbering
$ echo $3

```